commerce-machine-learning-impl/src/main/java/com/liferay/commerce/machine/learning/internal/forecast/BaseCommerceMLForecastServiceImpl.java

```diff
@@ -66,6 +70,31 @@
 public abstract class BaseCommerceMLForecastServiceImpl
 	<T extends CommerceMLForecast> {
 
+	protected T addCommerceMLForecast(T model) throws PortalException {
+		Document document = toDocumentModel(model);
+
+		document.addKeyword(Field.UID, String.valueOf(model.getForecastId()));
+
+		IndexDocumentRequest indexDocumentRequest = new IndexDocumentRequest(
+			commerceMLIndexer.getIndexName(model.getCompanyId()), document);
+
+		indexDocumentRequest.setType(_MAPPING_NAME);
+
+		IndexDocumentResponse indexDocumentResponse =
+			searchEngineAdapter.execute(indexDocumentRequest);
+
+		if ((indexDocumentResponse.getStatus() < 200) ||
+			(indexDocumentResponse.getStatus() > 300)) {
```

Comment:
we probably don't want to include 300 as an acceptable status, right?

Reply:
as far as I know correct answers are in the 2XX range only.

Reply:
I only ask because right now `300` passes this filter :). Should probably be `>= 300`.

Reply:
meh ... fixin'